addons/mail/static/src/views/fields/mail_server_configurator_selection/mail_server_configurator_selection.js

```diff
@@ -0,0 +1,76 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { SelectionField, selectionField } from "@web/views/fields/selection/selection_field";
+import { user } from "@web/core/user";
+import { useService } from "@web/core/utils/hooks";
+import { useState } from "@odoo/owl";
+
+export class MailServerConfiguratorSelection extends SelectionField {
```

Comment:
Following my earlier suggestion, would not be a field but a Component (I guess there is a selection component somewhere already?).
- Selection values is a method from the Component, overridden by gmail/outlook to add their values
- RPC to get the current value from a public method on the res.user record (should be accessible though the current model?)
- Selection change calls the same method as now, not much changes needed there

Should be doable and I really think it would be better than a field.